Extension/src/main.ts

```diff
@@ -413,33 +228,50 @@ function postInstall(installBlob: InstallBlob): Thenable<void> {
         }
     }
 
-    installBlob.telemetryProperties['osArchitecture'] = installBlob.info.architecture;
+    installBlob.telemetryProperties['osArchitecture'] = info.architecture;
 
     Telemetry.logDebuggerEvent("acquisition", installBlob.telemetryProperties);
 
+    return success;
+}
+
+async function postInstall(info: PlatformInformation): Promise<void> {
+    let channel: vscode.OutputChannel = util.getOutputChannel();
+
+    channel.appendLine("");
+    channel.appendLine("Finished installing dependencies");
+    channel.appendLine("");
+
+    const installSuccess: boolean = sendTelemetry(info);
+
     // If there is a download failure, we shouldn't continue activating the extension in some broken state.
-    if (installBlob.hasError) {
+    if (!installSuccess) {
         return Promise.reject<void>("");
-    }
+    } else {
+        const cpptoolsJsonFile: string = util.getExtensionFilePath("cpptools.json");
 
-    if (util.getDebuggerReloadLater()) {
-        util.showReloadPrompt();
-    }
+        try {
+            const exists: boolean = await util.checkFileExists(cpptoolsJsonFile);
+            if (exists) {
+                const cpptoolsString: string = await util.readFileText(cpptoolsJsonFile);
+                await cpptoolsJsonUtils.processCpptoolsJson(cpptoolsString);
+            }
+        } catch (error) {
+            // Ignore any cpptoolsJsonFile errors
+        }
 
-    return Promise.resolve();
-}
+        tempCommandRegistrar.activateLanguageServer();
```

Comment:
> tempCommandRegistrar.activateLanguageServer(); [](start = 7, length = 47)

should you await this before downloading the new file?

Reply:
Not a promise